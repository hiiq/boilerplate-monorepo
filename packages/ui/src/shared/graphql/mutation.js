import { gql } from '@apollo/client';

const USER_LOGIN = gql`
  mutation userLogin($input: UserLoginInput!) {
    userLogin(input: $input)
  }
`;

const USER_LOGOUT = gql`
  mutation userLogout {
    userLogout
  }
`;

const USER_REGISTER = gql`
  mutation userRegister($input: UserRegisterInput!) {
    userRegister(input: $input)
  }
`;

export const Mutation = {
  USER_LOGIN,
  USER_LOGOUT,
  USER_REGISTER,
};
