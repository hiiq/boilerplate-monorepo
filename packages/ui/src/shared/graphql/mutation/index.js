export * from './useUserLogin';
export * from './useUserLogout';
export * from './useUserRegister';
export * from './useUserSelfUpdate';
