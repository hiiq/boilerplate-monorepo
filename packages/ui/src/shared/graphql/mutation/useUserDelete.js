import { gql } from '@apollo/client';
import { useMutation } from '../useMutation';

const MUTATION_USER_DELETE = gql`
  mutation userDelete($id: ID!) {
    userDelete(id: $id)
  }
`;

const useUserDelete = options => useMutation(MUTATION_USER_DELETE, options);

export { useUserDelete };
