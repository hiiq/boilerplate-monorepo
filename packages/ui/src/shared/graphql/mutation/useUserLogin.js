import { gql } from '@apollo/client';
import { useMutation } from '../useMutation';

const MUTATION_USER_LOGIN = gql`
  mutation userLogin($input: UserLoginInput!) {
    userLogin(input: $input)
  }
`;

const useUserLogin = options => useMutation(MUTATION_USER_LOGIN, options);

export { useUserLogin };
