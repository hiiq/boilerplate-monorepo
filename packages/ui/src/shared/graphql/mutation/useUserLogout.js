import { gql } from '@apollo/client';
import { useMutation } from '../useMutation';

const MUTATION_USER_LOGOUT = gql`
  mutation userLogout {
    userLogout
  }
`;

const useUserLogout = options => useMutation(MUTATION_USER_LOGOUT, options);

export { useUserLogout };
