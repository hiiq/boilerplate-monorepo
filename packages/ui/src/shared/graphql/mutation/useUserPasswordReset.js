import { gql } from '@apollo/client';
import { useMutation } from '../useMutation';

const MUTATION_USER_PASSWORD_RESET = gql`
  mutation userPasswordReset($input: UserPasswordResetInput!) {
    userPasswordReset(input: $input)
  }
`;

/**
 * UserPasswordResetInput
 * @param id - ID!
 * @param passwordNew - String!
 * @param token - String!
 */
const useUserPasswordReset = (options) =>
  useMutation(MUTATION_USER_PASSWORD_RESET, options);

export { MUTATION_USER_PASSWORD_RESET, useUserPasswordReset };
