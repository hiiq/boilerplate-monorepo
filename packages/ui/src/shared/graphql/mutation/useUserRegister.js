import { gql } from '@apollo/client';
import { useMutation } from '../useMutation';

const MUTATION_USER_REGISTER = gql`
  mutation userRegister($input: UserRegisterInput!) {
    userRegister(input: $input)
  }
`;

/**
 * UserRegisterInput
 * @param email - String!
 * @param familyName - String!
 * @param givenName - String!
 * @param password - String!
 * @param role - String [Defaults to &#34;USER&#34;]
 * @param username - String!
 */
const useUserRegister = options => useMutation(MUTATION_USER_REGISTER, options);

export { useUserRegister };
