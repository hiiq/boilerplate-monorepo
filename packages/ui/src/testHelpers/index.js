import { cleanup, render } from '@testing-library/react';
import { curry } from 'ramda';
import React from 'react';
import { Providers } from 'modules/app';

export * from '@testing-library/react';

const customRender = curry((Component, defaultProps = {}, overrides = {}) =>
  render(<Component {...defaultProps} {...overrides} />, {
    wrapper: Providers,
  })
);

export { cleanup, customRender as render };
