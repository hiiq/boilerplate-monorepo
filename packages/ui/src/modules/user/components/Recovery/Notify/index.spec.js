import React from 'react';
import * as MutationHooks from 'shared/graphql/mutation/useUserRegister';
import { render } from 'testHelpers';
import { Notify } from '.';

jest.mock('shared/Button');
jest.mock('shared/Content');

jest.mock('./Form', () => ({
  Form: props => <x-Form {...props} />,
}));

describe('<Notify/>', () => {
  const defaultProps = {};

  const renderComponent = overrides =>
    render(<Notify {...defaultProps} {...overrides} />);

  beforeEach(() => {
    const useUserRegister = td.replace(MutationHooks, 'useUserRegister');

    td.when(useUserRegister()).thenReturn([
      jest.fn().mockName('userRegister'),
      { loading: false },
    ]);
  });

  test('renders properly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
