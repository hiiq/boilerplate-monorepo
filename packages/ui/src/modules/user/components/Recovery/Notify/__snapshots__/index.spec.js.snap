// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notify/> renders properly 1`] = `
<div>
  <x-header
    title="t(recoverAccount)"
  />
  <x-body
    hasbreadcrumbs="false"
  >
    <x-form
      user="[object Object]"
    />
  </x-body>
</div>
`;
